src/index.ts

```diff
@@ -315,3 +441,32 @@ export function generateSalt(byteCount = 32): string {
   );
   return b64encoded;
 }
+
+/**
+ * Checks if the provided key is an `EncryptionKey`.
+ *
+ * @param key - The key to check.
+ * @returns Whether or not the key is an `EncryptionKey`.
+ */
+export function isEncryptionKey(
+  key: CryptoKey | EncryptionKey,
+): key is EncryptionKey {
+  return (
+    'key' in key &&
+    'derivationOptions' in key &&
+    'algorithm' in key.derivationOptions &&
+    'params' in key.derivationOptions
+  );
+}
```

Comment:
Nit: Since we pass the result of `JSON.parse` into this function, would it be more accurate to say we don't know what this is? Also, the first part of this seems similar to `isExportedEncryptionKey`:

``` suggestion
export function isEncryptionKey(key: unknown): key is EncryptionKey {
  return (
    isExportedEncryptionKey(key) &&
    'algorithm' in key.derivationOptions &&
    'params' in key.derivationOptions
  );
}
```

Reply:
I made additional changes to this type guard, and also to `isExportedEncryptionKey`: these should not be similar, as one should be a `JsonWebKey` and the other a `CryptoKey`.

I also added checks for what's in `derivationOptions`